virtualizarr/manifests/manifest.py

```diff
@@ -165,99 +228,3 @@ def get_chunk_grid_shape(chunk_keys: Iterable[ChunkKey]) -> Tuple[int, ...]:
         max(indices_along_one_dim) + 1 for indices_along_one_dim in zipped_indices
     )
     return chunk_grid_shape
-
-
-def check_keys_form_grid(chunk_keys: Iterable[ChunkKey]):
-    """Check that the chunk keys collectively form a complete grid"""
```

Comment:
Lines 170-263 are what we get rid of by doing concatenation/stacking via the wrapped structured array.